File: gitdot-web/packages/api/src/endpoint/user/list-user-stars.ts
import { z } from "zod";
import { RepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListUserStarsRequest = z.object({});
export type ListUserStarsRequest = z.infer<typeof ListUserStarsRequest>;

export const ListUserStarsResponse = z.array(RepositoryResource);
export type ListUserStarsResponse = z.infer<typeof ListUserStarsResponse>;

export const ListUserStars = {
  path: "/user/{user_name}/stars",
  method: "GET",
  request: ListUserStarsRequest,
  response: ListUserStarsResponse,
} as const satisfies Endpoint;
export type ListUserStars = typeof ListUserStars;

File: gitdot-api/src/endpoint/user/list_user_stars.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{common::Page, repository::RepositoryResource},
};

pub struct ListUserStars;

impl Endpoint for ListUserStars {
    const PATH: &'static str = "/user/{user_name}/stars";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserStarsRequest;
    type Response = ListUserStarsResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListUserStarsRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListUserStarsResponse = Page<RepositoryResource>;
